### PR TITLE
[Tooling] Use a trusted CI agent for pipelines that require push access

### PR DESCRIPTION
**Pull request by iangmaia**
### Description
This PR uses the trusted agent helper script `use-bot-for-git` to use the GitHub bot for performing git push operations on CI.

### Testing Steps

We have to time the replacing of the existing deploy key to a read-only one to be able to test this PR, as it would break any ongoing release steps.
I'll add the tests to this PR before the Play Store Submission of version `19.4` (Jul 12th), then cherry-pick the changes to the `release/19.4` branch.

This PR will need anyway to be tested during the next release cycle of the project, when the affected pipelines should still be able to finish successfully.

### Follow-up
Once this PR is merged and making to a code freeze, we will need to revoke and replace in our setup the current deploy key with the new read-only one.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/224">19.4</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by AliSoftware — APPROVED** [review]
> I'll add the tests to this PR before the Play Store Submission of version 19.4 (Jul 12th), then cherry-pick the changes to the release/19.4 branch.
>
> This PR will need anyway to be tested during the next release cycle of the project, when the affected pipelines should still be able to finish successfully.
>
> Once this PR is merged and making to a code freeze, we will need to revoke and replace in our setup the current deploy key with the new read-only one.

👌

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/225">19.5</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: fastlane/Fastfile [hunk 1]
```diff
@@ -931,6 +931,18 @@ platform :android do
     )
   end
 
+  lane :test_git_push do |options|
```
Comment (dangermattic) [inline fastlane/Fastfile:934]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/woocommerce/woocommerce-android/blob/675d21289d9e0dfe4f729ec73ec2519976c702ec/fastlane/Fastfile#L934"></span>Lint/UnusedBlockArgument: Unused block argument - <code>options</code>. You can omit the argument if you don't care about it.</td>
    </tr>
  </tbody>
</table>

**Comment by iangmaia** [comment]
**Update**: as planned, I've deployed the read-only key and tested it on this repo:

- First, it failed to push, as expected: https://buildkite.com/automattic/woocommerce-android/builds/22153#01909e33-a641-47a3-8256-449263ad766d/112-199
- Once the `use-bot-for-git` script was sourced, it succeeded: https://buildkite.com/automattic/woocommerce-android/builds/22155#01909e35-9d94-4ade-92df-1ea1db1e2c63/120-175

**Comment by AliSoftware** [comment]
⚠️ @iangmaia   Seems this change [caused an issue during `finalize_release`](https://buildkite.com/automattic/woocommerce-android/builds/22253#0190a742-3fff-4d32-bce8-d7a30b13157f/151-255) today (see p1720792888491539/1719090144.437519-slack-C5ALGJYEP) because the `finalize_release` lane needs to download gradle from the internet, and new `tumblr-metal` agent used for this job now doesn't seem to have access.

**Pull request merged**